src/libraries/Native/Unix/System.Native/pal_networkstatistics.c

```diff
@@ -228,20 +240,49 @@ int32_t SystemNative_GetIcmpv4GlobalStatistics(Icmpv4GlobalStatistics* retStats)
 
 int32_t SystemNative_GetIcmpv6GlobalStatistics(Icmpv6GlobalStatistics* retStats)
 {
-    size_t oldlenp;
-
     assert(retStats != NULL);
 
-    struct icmp6stat systemStats;
-    oldlenp = sizeof(systemStats);
-    if (sysctlbyname("net.inet6.icmp6.stats", &systemStats, &oldlenp, NULL, 0))
+    size_t oldlenp = atomic_load(&icmp6statSize);
+    const char* sysctlName = "net.inet6.icmp6.stats";
+    void* buffer = malloc(oldlenp);
+    if (!buffer)
     {
         memset(retStats, 0, sizeof(Icmpv6GlobalStatistics));
+        errno = ENOMEM;
         return -1;
     }
 
-    uint64_t* inHist = systemStats.icp6s_inhist;
-    uint64_t* outHist = systemStats.icp6s_outhist;
+    int result = sysctlbyname(sysctlName, buffer, &oldlenp, NULL, 0);
+    if (result && errno == ENOMEM)
+    {
+        // We did not provide enough memory.
+        // macOS 11.0 added new member to icmp6stat so as FreeBSED reported changes between versions.
+        oldlenp = GetEstimatedSize(sysctlName);
+        buffer = realloc(buffer, oldlenp);
```

Comment:
If `realloc` fails and returns `NULL`, this leaks the `buffer` that `malloc` returned earlier.

Reply:
updated. (+1 one more place)